bindings/go/init.go

```diff
@@ -0,0 +1,21 @@
+package opendal
+
+import (
+	"github.com/ebitengine/purego"
+)
+
+var OPENDAL_LIB uintptr
+
+var opendal_operator_options_new func() uintptr
```

Comment:
cc @Ji-Xinyou, I will meet `segmentation violation` if I'm trying to pass the returning value of `opendal_operator_options_new` into `opendal_operator_new`. Maybe it's related to https://github.com/apache/incubator-opendal/issues/2462?

Reply:
I can try to fix that now